### PR TITLE
Documentation for creation of custom toolkit

### LINKED ISSUE #2639: Document how to create custom toolkits
**Is your feature request related to a problem? Please describe.**
The [toolkit guide](https://scala-cli.virtuslab.org/docs/guides/introduction/toolkit#other-toolkits) does not include an explanation on how to create a custom toolkit from scratch.

**Describe the solution you'd like**
An extra section in the doc describing how to create custom toolkits.

**Additional context**
Created out of https://github.com/VirtusLab/scala-cli/discussions/2637

### PR DESCRIPTION
**Pull request by yadavan88**
Based on [this discussion](https://github.com/VirtusLab/scala-cli/discussions/2637) and [this issue](https://github.com/VirtusLab/scala-cli/issues/2639), I added the information on how to create a custom toolkit. In the documentation, I referred to the sample toolkit created [here](https://github.com/yadavan88/lihaoyi-toolkit).

Please let me know if this looks fine or if any changes are required.

Thanks
Yadu

### CODE REVIEW

File: website/docs/guides/advanced/custom-toolkit.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+title: Custom Toolkit
+sidebar_position: 44
+---
+Similar to the Scala Toolkit and Typelevel toolkit, it is possible to create our own custom toolkit.
```
Comment (Gedochao) [inline website/docs/guides/advanced/custom-toolkit.md]:
```suggestion
Similar to the Scala Toolkit and Typelevel toolkit, it is possible to create your own, custom toolkit. 
```

File: website/docs/guides/advanced/custom-toolkit.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+sidebar_position: 44
+---
+Similar to the Scala Toolkit and Typelevel toolkit, it is possible to create our own custom toolkit. 
+Having a custom toolkit with common libraries can speed up the development using scala-cli. 
+
+Let's look at how we can create a new toolkit. 
+
+For example, to create a LiHaoyi ecosystem toolkit, we can name the file as `LiHaoyiToolkit.scala` and add the required libraries as dependency directives:
+
+```
```
Comment (Gedochao) [inline website/docs/guides/advanced/custom-toolkit.md]:
```suggestion
```scala
```

File: website/docs/guides/advanced/custom-toolkit.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+//> using publish.developer "yadavan88|Yadu Krishnan|https://github.com/yadavan88"
+//> using repository sonatype:public
+```
+
+The toolkit can be published locally using the command:
+```
+scala-cli --power publish local --cross LiHaoyiToolkit.scala publish-conf.scala
+```
+
+In the similar way, it is also possible to publish to a central repository. Refer to the [GitHub Action workflow](https://github.com/scala/toolkit/blob/main/.github/workflows/deploy.yaml) for more details.
```
Comment (Gedochao) [inline website/docs/guides/advanced/custom-toolkit.md]:
```suggestion
Similarly, it is also possible to publish to a central repository. Refer to the [GitHub Action workflow](https://github.com/scala/toolkit/blob/main/.github/workflows/deploy.yaml) for more details.
```

**Review by Gedochao — APPROVED** [review]
Looks good, thanks for the contribution!
I just commented on some nitpicks.

**Comment by yadavan88** [comment]
Thanks @Gedochao I applied your changes.

**Comment by Gedochao** [comment]
As the CI didn't trigger as it should for some reason, I rebased the PR from the latest `stable` branch, will merge it once the CI is green

**Pull request merged**